crml/cennzx-spot/src/types.rs

```diff
@@ -13,91 +13,157 @@
 //
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
-//!
-//! CENNZX-SPOT Types
-//!
 use cennznet_primitives::types::AssetId;
-use core::convert::TryInto;
-use uint::construct_uint;
-construct_uint! {
-	/// 128-bit unsigned integer.
-	pub struct U128(2);
+use codec::{Decode, Encode, HasCompact};
+use core::{
+	convert::{From, Into, TryFrom},
+	marker::PhantomData,
+};
+
+pub use primitive_types::U256 as HighPrecisionUnsigned;
+pub use u128 as LowPrecisionUnsigned;
+
+pub trait Scaled {
+	fn scale() -> LowPrecisionUnsigned;
 }
 
-construct_uint! {
-	/// 256-bit unsigned integer.
-	pub struct U256(4);
+/// Per millionth of unit price
+#[derive(Debug, Clone, Copy)]
+pub enum PerMill {}
```

Comment:
💭 was there a reason to prefer `enum PerMill {}` over a unit struct `struct PerMill{}`?

Reply:
I took the solution from the Rust by examples. Not sure. Just guessing under the hood it may be more efficient.

Reply:
Just found it's at the start of the rust book 🤦‍♂ https://doc.rust-lang.org/nomicon/exotic-sizes.html

Reply:
Nice notice!